src/search/pdbs/match_tree.h

```diff
@@ -22,11 +24,9 @@ namespace pdbs {
 */
 
 class MatchTree {
-    TaskProxy task_proxy;
+    const TaskProxy &task_proxy;
+    const Projection &projection;
```

Comment:
Having a reference here means that the `Projection` has to live longer than the `MatchTree`. This is probably fine but at least it should be documented.

Reply:
I think it would also be ok to copy it. If it just stores the pattern and multipliers, it should be light-weight enough.

Reply:
I don't like copying two vectors for no good reason. I'll document that Projection needs to live longer than MatchTree.

Reply:
In my opinion avoid a life-time dependency between two objects is a good reason to copy two vectors that we can expect to be small. Of course every vector could become so large it is a problem but specifically for patterns of PDBs, if the pattern is so large that copying is a problem, then we'll have other problems. We do not have many such life-time depencies in the code as far as I'm aware. There is one for the task that we accepted and work around by copying a shared pointer to the task everywhere where it should stay alive. There is another one with a log that recently caused problems.

Reply:
I agree, that one should avoid this. However, I think both AbstractOperator and MatchTree are classes "internal" to the PDB factory. They are visible only because they are in separate files (which I think is good). Projection is stored as part of the PDB factory, which also in the constructor computes the match tree and uses it to finish computing the PDB. Of course using a match tree in a different context is "dangerous", but it is just not intended to be used for something different. An alternative could be to not store the reference, but pass projection to both public methods of match treee: insertion and retrieval of operators. Would you prefer this?